PR for Frontend

AuroraLS3 opened the pull request:
To be merged when ready

AuroraLS3 committed:
Installed packages for frontend

axios
redux
react-redux
redux-thunk
react-router
bootstrap
reactstrap

AuroraLS3 committed:
Added CoreUI React Bootstrap template

AuroraLS3 committed:
remove old App.js

AuroraLS3 committed:
Added basic navigation & placeholder user head

AuroraLS3 committed:
Added rest of api calls to Dummy as a reminder

AuroraLS3 committed:
Added async await support with babel-polyfill

AuroraLS3 committed:
Axios getAll methods (users, backups, groups, plugins, worlds)

AuroraLS3 committed:
Redux store groundwork

AuroraLS3 committed:
Fixed class properties not working

AuroraLS3 committed:
Added views, begun work on Users view

AuroraLS3 committed:
Work on store initialization

AuroraLS3 committed:
Added a register page & redirected non-logged in users to login

AuroraLS3 committed:
Login now requests token from api

AuroraLS3 committed:
Replaced images

AuroraLS3 committed:
Merge remote-tracking branch 'remotes/origin/legacy' into frontend

AuroraLS3 committed:
Cleartext passtorage, Unregistered /register, abstracted PasswordStorage

AuroraLS3 committed:
401 returned instead of 400 when token expired

AuroraLS3 committed:
Added Image and source map responses to ResponseHandler

AuroraLS3 committed:
Fixed syntax errors due to line breaks

AuroraLS3 committed:
Added font responses

AuroraLS3 committed:
Fixed password file being overridden on enable

AuroraLS3 committed:
Updated npm plugins

AuroraLS3 committed:
Commented out register functionality from frontend

AuroraLS3 committed:
Attempted to fix password file loading on enable (still not working)

AuroraLS3 committed:
Removed register backend and frontend

AuroraLS3 committed:
Instruction modals to login page

AuroraLS3 committed:
Combined cleartext and hashed password storage

AuroraLS3 committed:
Fixed YamlPasswordStorage (Paths with dots not supported)

AuroraLS3 committed:
Added a logout button

AuroraLS3 committed:
Updated built frontend in the resources folder

AuroraLS3 committed:
Refactored Login Component

AuroraLS3 committed:
Started adding notifications

AuroraLS3 committed:
Fixed some Json error responses

AuroraLS3 committed:
Added notifications for logging in and out.

AuroraLS3 committed:
Multiple notifications can now be visible

AuroraLS3 committed:
Made expired token log the user out

Looks like there are some authentication issues that need to be solved
before users can log in again when using these changes.

AuroraLS3 committed:
Fixed Authentication issues

Caused by two different TokenVerifiers & axios.get doesn't take data
parameters.

AuroraLS3 committed:
Fixed some funky rendering stuff

AuroraLS3 committed:
Fixed Login-logout-login duplicating store contents

AuroraLS3 committed:
Fixed Logout sidebar staying open after logout click

AuroraLS3 committed:
Made sidebar use logged in user head instead of Dummy values

AuroraLS3 committed:
Flat permission display for Plugins-view

AuroraLS3 committed:
Moved view buttons to correct component

AuroraLS3 committed:
PluginJSONService Implementation

AuroraLS3 committed:
Swapped to better drag-list library

AuroraLS3 committed:
Improved Plugins view

AuroraLS3 committed:
Work on Worlds view

- Worlds, not yet sortable
- Permissions, sortable, negate button
- Inherited permissions, collapsible
- Add World button (no action yet)

AuroraLS3 committed:
World view functionality

AuroraLS3 committed:
Fixed dragged components disappearing when dragged

AuroraLS3 committed:
Made it possible to sort worlds, added world duplicate button, added possibility to rename world, added warning if world inherits itself

AuroraLS3 committed:
Prevented renaming to null when clicking cancel

AuroraLS3 committed:
Added eslint

AuroraLS3 committed:
Moved open state to be stored in redux

AuroraLS3 committed:
Prevented adding new or renaming to existing world

AuroraLS3 committed:
Copied and modified World view into Group view

AuroraLS3 committed:
Refactored World into reusable component

AuroraLS3 committed:
Implementation for User, Group, World JSONService get methods

AuroraLS3 committed:
Removed unused parts from Worlds-view world (Permissions)

AuroraLS3 committed:
Removed specific own permissions from group and user, transferred inside worlds without name.

AuroraLS3 committed:
Mapped new backend standard to old, Worlds to Group view

AuroraLS3 committed:
Removed permission inhertiance solving for groups due to performance

AuroraLS3 committed:
Made negated permissions stand out better

AuroraLS3 committed:
Removed tree view buttons from plugins-view

AuroraLS3 committed:
Made permission view copy buttons easier to understand

AuroraLS3 committed:
Prepared pagination component

AuroraLS3 committed:
Implemented rest of Group World functionality

AuroraLS3 committed:
Reducer redundant copying in worldsReducer

AuroraLS3 committed:
Fixed redirection to /users/ when logged in

AuroraLS3 committed:
Fixed a typo in Forgot pass modal

AuroraLS3 committed:
user service mapper

MrDienns commented:
You should open a pull request when it's ready for merge, not when it's still in progress.

Rsl1122 commented:
Pull requests are a tool.

It can be used to track progress and give feedback on code before the code gets merged into the main code base. If anyone practices code reviews, hardly any PR is ready when they're created.

MrDienns commented:
@Rsl1122 then you don't know how to use Git properly. Anywho, do what you want, I'm not stopping you, but clearly you've never worked in a proper organization before if you create pull requests every time you start working on something. A PR is suppose to be created when you think your code is finished and ready for merge, and then people can give feedback on it. Try opening a PR every time you start working on something in a company with 100 developers.

AuroraLS3 committed:
Added User component, implemented search and pagination for users

AuroraLS3 committed:
Added rest of the functionality to User components and User view, Added login page image

AuroraLS3 committed:
Linked user search to the redux store value

AuroraLS3 committed:
Local storage saving on change

AuroraLS3 committed:
Added warnings if a group doesn't exist

AuroraLS3 committed:
Save and discard changes buttons and button functionality

AuroraLS3 committed:
Fixed fonts by replacing font loader (file-loader -> url-loader)
https://stackoverflow.com/questions/34133808/webpack-ots-parsing-error-loading-fonts/34133809#34133809

AuroraLS3 committed:
Merge remote-tracking branch 'origin/frontend' into frontend

AuroraLS3 committed:
Backups view

AuroraLS3 committed:
Discard changes on backup restore, reverseMappers for saving

AuroraLS3 committed:
Partial BackupJSONServiceImpl, missing actual backup functionality

AuroraLS3 committed:
World JSON Saving, Better Json Exception handling

AuroraLS3 committed:
Fix backup removal rest api

AuroraLS3 committed:
Group JSON Saving

AuroraLS3 committed:
User JSON Saving

AuroraLS3 committed:
HTTP unsafe login notification

AuroraLS3 committed:
Added /pex dashboard-command with permissions.dashboard permission

Permission and command can be changed in the class if necessary

AuroraLS3 committed:
Updated frontend code in the resources folder

AuroraLS3 committed:
Moved dashboard command registration as it was causing error

AuroraLS3 committed:
Fixed � appearing when using html entities

AuroraLS3 committed:
Optimized Group, User, World saving (Now takes about 20 seconds for 50k users)

AuroraLS3 committed:
Access logging as well as config setting for it
